### ISSUE TITLE
Requesting license change to CC BY-NC-SA 4.0 to allow distribution of derived work

### ISSUE THREAD

**Issue by Avatair** [Glitchfiend/SereneSeasons#130]
My proposition is to change the license to **CC BY-NC-SA 4.0** for the non-API part of this mod and allow derivable work to be created and to be distributed by different authors. The motivation is, that this mod has been in a slow development since June 2018 due to limited resources of Glitchfiend, but has a huge unused potential for improvement. Improvement like **Coro's weather2 integration**, **season dependent plant textures**, **DynamicTrees integrations** are only a few examples.

The possibility of creating and distributing derived work will remove any bottlenecks for improvement of this mod. The concurrency of created forks will be a main driving force to make the mod better and leaving the choice for the end-user what fork he wants to use. Also the original mod authors can profit from this change by committing results from other forks to its own fork. Of course, the disadvantage for the original mod author is, that he transfers his control of the mod development to the community. He will consequently lose a full influence on it, because his code will represent only a fork among others.

**Special case**:
- **Existing mod integrations and API**: To avoid breaking compatibility with existing mods using the SereneSeasons API, like ToughAsNails or BiomesOPlenty, it is suggested, that the original API code is kept under the current **CC BY-NC-ND 4.0** license. It will be maintained and distributed as a separate jar only by Glitchfiend. Different forks should still be able to create their own additions to it without breaking the core API in any way.

**Comment by Forstride** [comment]
Yeaaaaah, that's not happening.  I rather enjoy having actual ownership over my mods.

Also 2 of the "improvements" you gave as examples are things that would be handled in those mods, not on our end.

**Comment by Forstride** [comment]
IDK if you're just salty about us still not merging your pull request yet, but this is by far the most extreme suggestion I've ever seen.

**Comment by Avatair** [comment]
It's up to you to accept this suggestion or not and it has nothing to do with my personal relation about the PR I submitted. Insulting you or your team was never my goal and I'm sorry for any misunderstanding or unintended harsh formulation in this suggestion. Also English is not my mothertongue. 

Your team is actually busy with the 1.13 porting of BoP and you have not enough time to maintain SS. These are things you guys stated in Discord and on Github. Judging by the github log of both project, this is confirmed. My conclusion is simply, that things would get easier for all of us, if you open this mod for the community so that everybody would help developing it. In the upper post I tried to outline advantages and disadvantages of this step as well as suggesting an architecture to avoid incompatibilities with the API. **This suggestion is one of the possible solutions to resolve the situation.** If you know a better solution, why not discussing it?

About my interest in this case, Part of my work depends on the functionalities of this mod and I require a clear base to build upon for two other season integration project I have. The first one is DynamicTrees which would have a dependency to my PR and the second is Growthcraft with a better crops support for plants being a multiblock structures, like trees. The API is clearly missing some functionalities in this fresh state. For example enabling the crop breaking feature would destroy a tree, if all blocks are registered as crops.

**Comment by Forstride** [comment]
You're asking us to give up ownership and control of our mod, that we've spent tons of time on already.  This isn't even worth debating for how asinine it is.

If you want to add features, make pull requests.  I get that you might be turned off from us not pulling yours yet, which is why I mentioned it, but that's just how it is.  I mean, people make tons of pull requests for Forge, a lot of which don't even get looked at _at all_, and that's just the risk you take when contributing to open/visible source software that you don't have control of.

**Comment by Avatair** [comment]
**PS:** Btw. my PR is about 50% of SS size. Just as a side note. So it might be debatable, whether you integrate my PR into SS or the other way round.

**Comment by Forstride** [comment]
But it wouldn't be my mod anymore if this were to happen, so there's no "honor" in that.

If other mods "depend" on it, guess what?  Their authors can make pull requests!  That's what they're meant for.

**Comment by Avatair** [comment]
Or guess what. Search for a replacement.

**Comment by Forstride** [comment]
Also you're acting as if we've abandoned the mod and never plan on updating it again.  That's not the case at all.  I even told you originally when you made your chunk simulation PR that it probably wouldn't be until we update to 1.13 that we look into it and implement it.

**Comment by Avatair** [comment]
Well. Updating to 1.13 is up to you then. Judging by the github log this mod is barely kept up-to-date. The PR is a year old considering the version for TAN, too. That means something ...

**Comment by Forstride** [comment]
I just did a TON of stuff last month and in November, and you're gonna say it's barely kept up-to-date?  Sorry we're not adding major features every single day, but when we have BOP's 1.13 update to worry update, and our own personal lives, it's not exactly gonna be like that...

**Comment by Avatair** [comment]
Well, I'm tired of arguing about a clear case. Take a look at BoP and then at SS about commitment rate. Convince yourself. A TON of work has been done for BoP not for SS ... You guys may search for more people to handle extra work if you are not wiling to open the mod for community. I'll search for an alternative for myself and include my PR there which is btw. Half the size of SS.

**Comment by Forstride** [comment]
Well, yes, BOP is our most important mod with 17 million+ downloads compared to the less than 200k that Serene Seasons has.  No shit we're going to give it more attention.

This is a hobby for us, it's not our job, it's not something that we devote our lives to.  Sorry your work attitude doesn't mesh with ours, but thinking I'm just gonna give you our mod because of it is the dumbest fucking thing I've ever heard of.

I tried to be nice, I tried to be reasonable and explain things to you, but you're acting like an entitled asshole over it.
